app/src/main/java/com/mobileproto/hireddit/hireddit/visuals/SpeechFragment.java

```diff
@@ -107,6 +117,45 @@ public void onClick(View v) {
                     doListen();
             }
         });
+
+        muteButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                voiceMode();
+            }
+        });
+
+        volumeOnButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                quietMode();
+            }
+        });
+
+        speechTextDisplay.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                typeMode();
+            }
+        });
+
+        TextInputDisplay.setOnEditorActionListener(
+                new EditText.OnEditorActionListener() {
+                    @Override
+                    public boolean onEditorAction(TextView v, int actionId, KeyEvent event) {
+                        if (actionId == EditorInfo.IME_ACTION_DONE) {
+                            InputMethodManager mgr = (InputMethodManager) getActivity().getSystemService(Context.INPUT_METHOD_SERVICE);
+                            mgr.hideSoftInputFromWindow(TextInputDisplay.getWindowToken(), 0);
+                            ArrayList<String> TextInput = new ArrayList<String>();
+                            TextInput.add(0, TextInputDisplay.getText().toString());
+                            speechResultCallback(TextInput);
```

Comment:
does speechResultCallback only take an arraylist? we probably should make it a string since i think this method only ever uses the first element right